packages/training/python/src/training/scenario_pool.py

```
@@ -0,0 +1,1072 @@
+
+    def _generate_contextual_news(self, markets: List[MarketState]) -> List[NewsItem]:
+        """Generate news relevant to the markets"""
+        news = []
+        
+        for market in markets[:3]:
+            # Extract key terms from question
+            question_lower = market.question.lower()
+            
+            if "btc" in question_lower or "bitcoin" in question_lower:
+                news.append(NewsItem(
+                    headline=f"Bitcoin Technical Analysis: Key Levels to Watch",
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)